packages/esm-service-queues-app/src/queue-table/transitions/transfer-queue-dialog.component.tsx

```diff
@@ -0,0 +1,186 @@
+
+  const statuses = selectedQueue?.allowedStatuses;
+  const hasNoStatusesConfigured = selectedQueue && statuses.length == 0;
+  const priorities = selectedQueue?.allowedPriorities;
+  const hasNoPrioritiesConfigured = selectedQueue && priorities.length == 0;
+
+  const setSelectedQueueUuid = (selectedQueueUuid: string) => {
+    const newSelectedQueue = queues.find((q) => q.uuid == selectedQueueUuid);
+    const { allowedStatuses, allowedPriorities } = newSelectedQueue;
+    const newQueueHasCurrentStatus = allowedStatuses.find((s) => s.uuid == formState.selectedStatus);
+    const newQueueHasCurrentPriority = allowedPriorities.find((s) => s.uuid == formState.selectedPriority);
+    setFormState({
+      selectedQueue: selectedQueueUuid,
+      selectedStatus: newQueueHasCurrentStatus ? formState.selectedStatus : allowedStatuses[0]?.uuid,
+      selectedPriority: newQueueHasCurrentPriority ? formState.selectedPriority : allowedPriorities[0]?.uuid,
+    });
+  };
+
+  const setSelectedPriorityUuid = (selectedPriorityUuid: string) => {
+    setFormState({ ...formState, selectedPriority: selectedPriorityUuid });
+  };
+
+  const setSelectedStatusUuid = (selectedStatusUuid: string) => {
+    setFormState({ ...formState, selectedStatus: selectedStatusUuid });
+  };
+
+  const transferQueue = (e) => {
+    e.preventDefault();
+
+    transferQueueEntry(queueEntry.uuid, formState.selectedQueue, formState.selectedStatus, formState.selectedPriority)
+      .then(({ status }) => {
+        if (status === 200) {
+          showSnackbar({
+            isLowContrast: true,
+            title: t('queueEntryTransferred', 'Queue entry transferred'),
+            kind: 'success',
+            subtitle: t('queueEntryTransferredSuccessfully', 'Queue entry transferred successfully'),
+          });
+          mutateQueueEntries();
+          closeModal();
+        } else {
+          throw { message: t('unexpectedServerResponse', 'Unexpected Server Response') };
+        }
+      })
+      .catch((error) => {
+        showSnackbar({
+          title: t('queueEntryTranferFailed', 'Error transferring queue entry'),
+          kind: 'error',
+          subtitle: error?.message,
+        });
+      });
+  };
+
+  return (
+    <div>
+      <Form onSubmit={transferQueue}>
+        <ModalHeader closeModal={closeModal} title={t('transferToNextQueue', 'Transfer to next queue')} />
+        <ModalBody>
+          <div className={styles.modalBody}>
+            <h5>{queueEntry.display}</h5>
+          </div>
+
+          <section className={styles.section}>
+            <div className={styles.sectionTitle}>{t('serviceQueue', 'Service queue')}</div>
+            <Select
+              labelText={t('selectQueue', 'Select a queue')}
+              id="queue"
+              invalidText="Required"
+              value={formState.selectedQueue}
+              onChange={(event) => setSelectedQueueUuid(event.target.value)}>
+              {queues?.map((queue) => (
```

Comment:
After discussing with @mseaton more, we are consolidating transitions (of queues and/or status) into a single form. I think this means we should default to the currently selected queue (in case the user only wants to transition the status).